src/zip/mod.rs

```diff
@@ -92,7 +92,7 @@ where
 {
     type Output = ArrayView<'a, A, E::Dim>;
     fn broadcast_unwrap(self, shape: E) -> Self::Output {
-        let res: ArrayView<'_, A, E::Dim> = (&self).broadcast_unwrap(shape.into_dimension());
+        let res: ArrayView<'_, A, E::Dim> = self.broadcast_unwrap(shape.into_dimension());
```

Comment:
Linter is wrong and it shouldn't be changed, it looks like. local suppression possible?

Reply:
yeah, I believe so, at least it seems that way from this [SO post](https://stackoverflow.com/questions/55402812/how-to-disable-a-clippy-lint-for-a-single-line-block)

Reply:
I suppressed this lint for the specified function